aim/sdk/adapters/fastai.py

```diff
@@ -0,0 +1,132 @@
+from logging import getLogger
+from typing import Optional
+from aim.sdk.run import Run
+
+from aim.ext.resource.configs import DEFAULT_SYSTEM_TRACKING_INT
+
+try:
+    from fastai.learner import Callback
+    from fastcore.basics import store_attr, detuplify, ignore_exceptions
+    from fastai.callback.hook import total_params
+except ImportError:
+    raise RuntimeError(
+        'This contrib module requires fastai to be installed. '
+        'Please install it with command: \n pip install fastai'
+    )
+
+logger = getLogger(__name__)
+
+
+class AimCallback(Callback):
+    def __init__(self,
+                 repo: Optional[str] = None,
+                 experiment_name: Optional[str] = None,
+                 system_tracking_interval: Optional[int] = DEFAULT_SYSTEM_TRACKING_INT,
+                 log_system_params: bool = True,
+                 ):
+        store_attr()
+        self._run = None
+        self._run_hash = None
+
+    @property
+    def experiment(self) -> Run:
+        if not self._run:
+            self.setup()
+        return self._run
+
+    def setup(self):
+        if not self._run:
+            if self._run_hash:
+                self._run = Run(
+                    self._run_hash,
+                    repo=self.repo,
+                    system_tracking_interval=self.system_tracking_interval,
+                )
+            else:
+                self._run = Run(
+                    repo=self.repo,
+                    experiment=self.experiment_name,
+                    system_tracking_interval=self.system_tracking_interval,
+                    log_system_params=self.log_system_params,
+                )
+                self._run_hash = self._run.hash
+
+        # Log config parameters
+        configs_log = self.gather_args()
```

Comment:
what about moving the arguments gathering/setting run h-params directly into `before_fit`?
`setup` is meant to create/re-create the inner Run object.

Reply:
Should we gather the arguments twice? Once in `experiment` and one in `before_fit`.